applications/portal/frontend/src/components/CustomAccordionSummary.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {
+    Box,
+    AccordionSummary,
+    Switch,
+    FormControlLabel,
+    Popover
+} from '@material-ui/core';
+
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import UP_ICON from "../assets/images/icons/up.svg";
+import { POPULATION_FINISHED_STATE } from "../utilities/constants";
+import { getRGBAFromHexAlpha, getRGBAString } from "../utilities/functions";
+import ColorPicker from "./common/ColorPicker";
+import SwitchLabel from "./common/SwitchLabel";
+import { TrailIcon, TrailEndIcon } from './icons';
+
+
+const POPULATION_ICONS_OPACITY = 0.4
+
+const CustomAccordionSummary = ({
+    id,
+    data,
+    pId,
+    isExpanded,
+    populations,
+    isParent,
+    isChild,
+    handlePopulationSwitch,
+    handlePopulationColorChange,
+    hasEditPermission,
+
+}) => {
+    const [popoverAnchorEl, setPopoverAnchorEl] = React.useState(null);
+    const [selectedPopoverId, setSelectedPopoverId] = React.useState(null);
+
+
+    const handlePopoverClick = (event, id) => {
+        if (!hasEditPermission) {
```

Comment:
nitpick: @Aiga115 afaik this guard isn't needed, this is (or should) already handled in the rendering part